app/change_sets/change_set.rb

```diff
@@ -135,19 +129,22 @@ def fix_multivalued_keys
     @_changes = Disposable::Twin::Changed::Changes.new
   end
 
+  # Populator Documentation: https://trailblazer.to/2.0/gems/reform/populator.html
   def populate_nested_collection(fragment:, as:, collection:, index:, **)
     property_klass = model_type_for(property: as)
+
+    # Find the nested object we want to change
     item = collection.find { |x| x.id.to_s == (fragment["id"] || fragment[:id]) }
-    if item
-      if delete_fragment?(fragment)
-        collection.delete_at(index)
-        return skip!
-      else
-        item
-      end
-    elsif delete_fragment?(fragment)
+
+    if delete_fragment?(fragment)
+      # If the object should be deleted, then delete it if it exists
+      collection.delete_at(index) if item
```

Comment:
Suggestion: Since you have the tests set up now, what is `index` when the item doesn't exist? Is it nil? If so, maybe we can key off that instead of finding the item at all?

Reply:
When the item doesn't exist, ``index`` isn't nil but the index where we should add the item. In a lot of our test cases, that index is zero.